rotkehlchen/premium/premium.py

```diff
@@ -353,6 +355,9 @@ def delete_device(self, device_id: str) -> None:
         May raise:
         - RemoteError
         """
+        if device_id == machineid.hashed_id(self.username):
+            raise RemoteError('Cannot delete the current device')
```

Comment:
Why RemoteError? Just because it's the only error caught? It does not make much sense as this is not a remote Error. This is a user error. I would raise a different error and also return a different status. Like a 409 conflict for when this happens as it's asking for something that should not be allowed. And validation happens locally, not remote.